### ISSUE TITLE
M2.4.4 + php 8.1 static:content:deploy -f

### ISSUE THREAD

**Issue by leonhelmus** [magento/magento2#36020]
### Preconditions and environment

- Magento version 2.4.4 or higher
- php 8.1
- Create a theme and have Luma as a parant.
- Have no additional files except for some xml files

### Steps to reproduce

Go to CLI and do the command:
bin/magento setup:static-content:deploy --area frontend --theme Custom/default --language nl_NL  -f

### Expected result

It generates the static content.

### Actual result

It gives an error: 
Error happened during deploy process: Deprecated Functionality: pathinfo(): Passing null to parameter #1 ($path) of type string is deprecated in.

### Additional information

To resolve this you need to apply this patch:
```
diff --git a/vendor/magento/framework/App/Utility/Files.php b/vendor/magento/framework/App/Utility/Files.php
--- a/vendor/magento/framework/App/Utility/Files.php
+++ b/vendor/magento/framework/App/Utility/Files.php	(date 1661420041266)
@@ -958,7 +958,7 @@
                         $themePackage->getVendor() . '/' . $themePackage->getName(),
                         null,
                         null,
-                        null,
+                        '',
                         null
                     ];
                 }
```

The fifth parameter needs to be a string otherwise pathinfo will fail since that now only accepts string and not NULL anymore.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Lima** [comment]
Issue got reproduced in `2.4 develop`.

**Description:**
When we hit the bin/magento setup:static-content:deploy  --area frontend --theme Customtheme/mytheme --language nl_NL -f  command we are getting error.
 
**Pre-requisite:**

- Magento version 2.4 develop
- php 8.1
- Created custom theme with Luma as a parent

 
**Steps to reproduce:**
1.Created a custom theme and Luma as a parent
2. Hit below command in terminal
`bin/magento setup:static-content:deploy --area frontend --theme Customtheme/mytheme --language nl_NL -f`

**Expected result:**	
1.It generates the static content.

**Actual result:**
1.Error happened during deploy process: Deprecated Functionality: pathinfo(): Passing null to parameter #1 ($path) of type string is deprecated in `/var/www/html/ac36020/lib/internal/Magento/Framework/View/Asset/PreProcessor/FileNameResolver.php on line 42`

**Screenshots:**
![pasted image 0](https://user-images.githubusercontent.com/60198022/187206045-65d8e2a0-8b0e-485d-b5da-e58a14889ef5.png)

Hence,confirming the issue.

Thanks

**Comment by texboy** [comment]
@magento I am working on this

**Comment by engcom-Delta** [comment]
Hi @leonhelmus,

The issue has been fixed in 12a9eacaf57baf1dbd71b22cbd9ada32f3ffa478, and is not reproducible in Magento2.4-develop now. Hence we are closing this issue.

Thanks
